File: comb-manager-school-web/src/main/webapp/js/service/brandService.js
// 定义服务层:
app.service("brandService",function($http){
	this.findAll = function(){
		return $http.get("../brand/findAll.do");
	}
	
//	this.findPage = function(page,size){
//		return $http.get("../brand/findPage.do?page="+page+"&rows="+size);
//	}
	
	this.add = function(entity){
		return $http.post("../brand/add.do",entity);
	}
	
	this.update=function(entity){
		return $http.post("../brand/update.do",entity);
	}
	
	this.findOne=function(id){
		return $http.get("../brand/findOne.do?id="+id);
	}
	
	this.dele = function(ids){
		return $http.get("../brand/delete.do?ids="+ids);
	}
	
	this.search = function(page,rows,searchEntity){
		return $http.post("../brand/search.do?page="+page+"&rows="+rows,searchEntity);
	}
	
	this.selectOptionList = function(){
		return $http.get("../brand/selectOptionList.do");
	}
});

File: comb-publish-interface/src/main/java/com/comb/publish/service/BrandService.java
package com.comb.publish.service;

import java.util.List;
import java.util.Map;

import com.comb.pojo.TbBrand;

import entity.PageResult;

/**
 * 品牌接口
 * @author WangQingBang
 *
 */
public interface BrandService {
	
	public List<TbBrand> findAll();
	
//	/**
//	 * 品牌分页
//	 * @param pageNum	当前页面
//	 * @param pageSize	每页记录数
//	 * @return
//	 */
//	public PageResult findPage(int pageNum, int pageSize);
	/**
	 * 增加
	 * @param brand
	 */
	public void add(TbBrand brand);
	/**
	 * 根据id查询实体
	 * @param id
	 * @return
	 */
	public TbBrand findOne(Long id);
	/**
	 * 修改
	 * @param brand
	 */
	public void update(TbBrand brand);
	/**
	 * 复选多项删除
	 * @param ids
	 */
	public void delete(Long[] ids);
	/**
	 * 多条件查询，利用方法重载，新写一个方法
	 * @param brand
	 * @return
	 */
	public PageResult findPage(TbBrand brand, int pageNum, int pageSize);
	/**
	 * 返回下拉列表数据
	 * @return
	 */
	public List<Map> selectOptionList();
}
